cmd/drone-server/drone.go

```diff
@@ -236,6 +244,14 @@ func main() {
 		auth.POST("", server.GetLogin)
 	}
 
+	redirects := r.Group("/redirect")
+	{
+		redirects.Use(server.SetDatastore(store))
+		redirects.Use(server.SetRepo())
+
+		redirects.GET("/:owner/:name/commit/:sha", server.RedirectSha)
```

Comment:
@Bugagazavr 
Jenkins integration is only in Enterprise edition, but source code is publicly visible. So you can check, how it is implemented:
https://gitlab.com/gitlab-org/gitlab-ee/blob/master/app/models/project_services/jenkins_service.rb